### ISSUE TITLE
spurious CONFLICTs detected

### ISSUE THREAD

**Issue by moench-tegeder** [2ndQuadrant/pglogical#24]
Setup: two PostgreSQL instances db1 (port 5433) and db2 (port 5434). Data is the pgbench database with an added primary key on pgbench_history (dbname is bench). The db2 instance's "bench" will be subscriber to the db1 instance's "bench".

No matter wether I set up db2 via `pglogical_create_subscriber` or via SQL `pglogical.create_subscription(..., synchronize_structure = 't', synchronize_data := 't')` an update to `pgbench_accounts` (`update pgbench_accounts update set abalance = abalance + 1 where aid = 1;`) results in a CONFLICT log message:
`LOG:  CONFLICT: remote UPDATE on relation public.pgbench_accounts. Resolution: apply_remote.`
Subsequent UPDATEs to the same row do not yield CONFLICT messages. (Other tables are affected, too - but this is the straight forward reproducer).

Versions: 
- `PostgreSQL 9.5.3 on x86_64-unknown-freebsd10.3, compiled by gcc (FreeBSD Ports Collection) 4.8.5, 64-bit`, behaviour also reproducable on Linux
- pg_logical 1.1.2 (as per pg_extension), also reproducable with currently released Debian packages

**Comment by 1803** [comment]
I'd try either synchronizing or segmenting the sequences on the table.
```
alter sequence bench_id_seq start 5000000000000000000;
alter sequence bench_id_seq restart;
```
on slave;

**Comment by PJMODOS** [comment]
It's possible that we mistakenly report CONFLICTS when track_commit_timestamps is off

**Comment by moench-tegeder** [comment]
@1803: it's the pgbench database, there's no sequence involved.

**Comment by moench-tegeder** [comment]
Unless I'm totally wrong here, `track_commit_timestamp` is on. I'm attaching my postgresql.conf, and `SHOW track_commit_timestamp` confirms my setting.

**Comment by 1803** [comment]
@moench-tegeder  

> database with an added primary key on pgbench_history 

This part was a bit misleading.
You could try removing PK and creating full replica identity and see if that helps.

`alter table bench set replica identity full;`

Refer to alter table doc: https://www.postgresql.org/docs/current/static/sql-altertable.html

**Comment by moench-tegeder** [comment]
this thing relies on file extensions like it's 1995. so here's the postgresql.conf.

`port = 5433
max_connections = 20
max_replication_slots = 5
max_wal_senders = 10
wal_level = logical
hot_standby = on
max_wal_size = 512MB
min_wal_size = 160MB
checkpoint_warning = 0
checkpoint_timeout = 600
track_commit_timestamp = on
log_destination = stderr
logging_collector = on
log_directory = pg_log
log_min_messages = notice
shared_preload_libraries = 'pglogical,pg_stat_statements'`

**Comment by moench-tegeder** [comment]
@1803: look again - the history table is not touched here, I'm doing a single (ok, two) UPDATEs to the pgbench_accounts table. The primary key is just for convenience so I can create the subscribe in as few steps as possible.

**Comment by thedumbtechguy** [comment]
I've observed the same thing in my setup. There were just enough of them that I assumed it was normal. 
When tailing the logs, some updates showed the conflict, some didn't.

I'm using an md5 value for the primary key.

**Comment by PJMODOS** [comment]
This should be fixed in 2.0.
